File: js/generators_1.js
function* generator(x=1) {
    while(1){
      let res = yield x++;
      if(res!==undefined) x=res;
    }
  }
  //res === undefined, если в next() не передано аргумента,
  // либо это самый первый next()

  ////////////
  function* generator(i = 1) {
    while(true){
      var value = yield i++;
      i = value || i;
    }
  }

  /////////////
  function* generator() {
    let a = 1, b;
    while(true){
      b = yield b || a++;
    }
  }

File: java/generators_1.java
public class Generator{
  private int x=0;
  public int next(){
    return ++x;
  }
  public int next(int y){
    return y;
  }
}

////////////
public class Generator {

  private int n = 0;
  
  public int next() { return ++n; }
  public int next(final int n) { return this.n = n; }
}

///////////
class Generator {
  private int value;

  int next(Integer... reset) {
    return reset.length > 0 ? reset[0] : ++value;
  }
}